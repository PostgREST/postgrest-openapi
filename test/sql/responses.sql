-- defines a default error response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'defaultError');

-- defines an empty response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'empty');

-- Tables
-- Non-empty response body

-- defines an application/json response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.products'->'content'->'application/json');

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.products'->'content'->'application/vnd.pgrst.object+json');

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');

-- defines a text/csv response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.products'->'content'->'text/csv');

-- Empty or non-empty response body

-- defines an application/json response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.products'->'content'->'application/json');

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.products'->'content'->'application/vnd.pgrst.object+json');

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');

-- defines a text/csv response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.products'->'content'->'text/csv');

-- Views
-- Non-empty response body

select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.big_products'->'content'->'application/json');

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.big_products'->'content'->'application/vnd.pgrst.object+json');

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.big_products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');

-- defines a text/csv response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'notEmpty.big_products'->'content'->'text/csv');

-- defines a non-empty response body for non auto-updatable views
select postgrest_openapi_spec('{test}')->'components'->'responses' ? 'notEmpty.non_auto_updatable' as value;

-- Empty or non-empty response body

-- defines an application/json response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.big_products'->'content'->'application/json');

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.big_products'->'content'->'application/vnd.pgrst.object+json');

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.big_products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');

-- defines a text/csv response
select jsonb_pretty(postgrest_openapi_spec('{test}')->'components'->'responses'->'mayBeEmpty.big_products'->'content'->'text/csv');

-- does not define an empty or non-empty response body for non auto-updatable views
select postgrest_openapi_spec('{test}')->'components'->'responses' ? 'mayBeEmpty.non_auto_updatable' as value;
