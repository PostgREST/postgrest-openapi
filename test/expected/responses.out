-- defines a default error response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'defaultError');
                jsonb_pretty                
--------------------------------------------
 {                                         +
     "content": {                          +
         "application/json": {             +
             "schema": {                   +
                 "type": "object",         +
                 "properties": {           +
                     "code": {             +
                         "type": "string"  +
                     },                    +
                     "hint": {             +
                         "type": "string"  +
                     },                    +
                     "details": {          +
                         "type": "string"  +
                     },                    +
                     "message": {          +
                         "type": "string"  +
                     }                     +
                 }                         +
             }                             +
         }                                 +
     },                                    +
     "description": "Default error reponse"+
 }
(1 row)

-- Tables
-- GET operation object
-- defines an application/json response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.products'->'content'->'application/json');
                    jsonb_pretty                     
-----------------------------------------------------
 {                                                  +
     "schema": {                                    +
         "type": "array",                           +
         "items": {                                 +
             "$ref": "#/components/schemas/products"+
         }                                          +
     }                                              +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.products'->'content'->'application/vnd.pgrst.object+json');
                  jsonb_pretty                   
-------------------------------------------------
 {                                              +
     "schema": {                                +
         "$ref": "#/components/schemas/products"+
     }                                          +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');
                  jsonb_pretty                   
-------------------------------------------------
 {                                              +
     "schema": {                                +
         "$ref": "#/components/schemas/products"+
     }                                          +
 }
(1 row)

-- defines a text/csv response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.products'->'content'->'text/csv');
       jsonb_pretty        
---------------------------
 {                        +
     "schema": {          +
         "type": "string",+
         "format": "csv"  +
     }                    +
 }
(1 row)

-- POST operation object
-- defines an application/json response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.products'->'content'->'application/json');
                        jsonb_pretty                         
-------------------------------------------------------------
 {                                                          +
     "schema": {                                            +
         "oneOf": [                                         +
             {                                              +
                 "type": "array",                           +
                 "items": {                                 +
                     "$ref": "#/components/schemas/products"+
                 }                                          +
             },                                             +
             {                                              +
                 "type": "string"                           +
             }                                              +
         ]                                                  +
     }                                                      +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.products'->'content'->'application/vnd.pgrst.object+json');
                      jsonb_pretty                       
---------------------------------------------------------
 {                                                      +
     "schema": {                                        +
         "oneOf": [                                     +
             {                                          +
                 "$ref": "#/components/schemas/products"+
             },                                         +
             {                                          +
                 "type": "string"                       +
             }                                          +
         ]                                              +
     }                                                  +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');
                      jsonb_pretty                       
---------------------------------------------------------
 {                                                      +
     "schema": {                                        +
         "oneOf": [                                     +
             {                                          +
                 "$ref": "#/components/schemas/products"+
             },                                         +
             {                                          +
                 "type": "string"                       +
             }                                          +
         ]                                              +
     }                                                  +
 }
(1 row)

-- defines a text/csv response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.products'->'content'->'text/csv');
       jsonb_pretty        
---------------------------
 {                        +
     "schema": {          +
         "type": "string",+
         "format": "csv"  +
     }                    +
 }
(1 row)

-- Views
-- GET operation object
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.big_products'->'content'->'application/json');
                      jsonb_pretty                       
---------------------------------------------------------
 {                                                      +
     "schema": {                                        +
         "type": "array",                               +
         "items": {                                     +
             "$ref": "#/components/schemas/big_products"+
         }                                              +
     }                                                  +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.big_products'->'content'->'application/vnd.pgrst.object+json');
                    jsonb_pretty                     
-----------------------------------------------------
 {                                                  +
     "schema": {                                    +
         "$ref": "#/components/schemas/big_products"+
     }                                              +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.big_products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');
                    jsonb_pretty                     
-----------------------------------------------------
 {                                                  +
     "schema": {                                    +
         "$ref": "#/components/schemas/big_products"+
     }                                              +
 }
(1 row)

-- defines a text/csv response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'get.big_products'->'content'->'text/csv');
       jsonb_pretty        
---------------------------
 {                        +
     "schema": {          +
         "type": "string",+
         "format": "csv"  +
     }                    +
 }
(1 row)

-- defines a GET operation object for non auto-updatable views
select get_postgrest_openapi_spec('{test}')->'components'->'responses' ? 'get.non_auto_updatable' as value;
 value 
-------
 t
(1 row)

-- POST operation object
-- defines an application/json response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.big_products'->'content'->'application/json');
                          jsonb_pretty                           
-----------------------------------------------------------------
 {                                                              +
     "schema": {                                                +
         "oneOf": [                                             +
             {                                                  +
                 "type": "array",                               +
                 "items": {                                     +
                     "$ref": "#/components/schemas/big_products"+
                 }                                              +
             },                                                 +
             {                                                  +
                 "type": "string"                               +
             }                                                  +
         ]                                                      +
     }                                                          +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.big_products'->'content'->'application/vnd.pgrst.object+json');
                        jsonb_pretty                         
-------------------------------------------------------------
 {                                                          +
     "schema": {                                            +
         "oneOf": [                                         +
             {                                              +
                 "$ref": "#/components/schemas/big_products"+
             },                                             +
             {                                              +
                 "type": "string"                           +
             }                                              +
         ]                                                  +
     }                                                      +
 }
(1 row)

-- defines an application/vnd.pgrst.object+json;nulls=stripped response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.big_products'->'content'->'application/vnd.pgrst.object+json;nulls=stripped');
                        jsonb_pretty                         
-------------------------------------------------------------
 {                                                          +
     "schema": {                                            +
         "oneOf": [                                         +
             {                                              +
                 "$ref": "#/components/schemas/big_products"+
             },                                             +
             {                                              +
                 "type": "string"                           +
             }                                              +
         ]                                                  +
     }                                                      +
 }
(1 row)

-- defines a text/csv response
select jsonb_pretty(get_postgrest_openapi_spec('{test}')->'components'->'responses'->'post.big_products'->'content'->'text/csv');
       jsonb_pretty        
---------------------------
 {                        +
     "schema": {          +
         "type": "string",+
         "format": "csv"  +
     }                    +
 }
(1 row)

-- does not define a POST operation object for non auto-updatable views
select get_postgrest_openapi_spec('{test}')->'components'->'responses' ? 'post.non_auto_updatable' as value;
 value 
-------
 f
(1 row)

